test_unstructured_inference/test_elements.py

```diff
@@ -191,6 +195,47 @@ def test_intersection_over_min(
     )
 
 
+def test_grow_region_to_match_region():
+    from unstructured_inference.inference.elements import Rectangle, grow_region_to_match_region
+
+    a = Rectangle(1, 1, 2, 2)
+    b = Rectangle(1, 1, 5, 5)
+    grow_region_to_match_region(a, b)
+    assert a == Rectangle(1, 1, 5, 5)
+
+
+def test_LocationlessLayoutElement():
+    text = "Testing text"
+    type = "Type"
+    e = LocationlessLayoutElement(text, type)
+    assert e.to_dict() == {"text": text, "type": type}
+
+
+@pytest.mark.parametrize(
+    ("rect1", "rect2", "expected"),
+    [
+        (elements.Rectangle(0, 0, 5, 5), elements.Rectangle(3, 3, 5.1, 5.1), True),
+        (elements.Rectangle(0, 0, 5, 5), elements.Rectangle(3, 3, 5.2, 5.2), True),
+        (elements.Rectangle(0, 0, 5, 5), elements.Rectangle(7, 7, 10, 10), False),
+    ],
+)
+def test_is_almost_subregion_of(rect1, rect2, expected):
+    assert expected == rect2.is_almost_subregion_of(rect1)
+
+
+@pytest.mark.parametrize(
+    ("rect1", "rect2"),
+    [
+        (elements.Rectangle(0, 0, 5, 5), elements.Rectangle(3, 3, 6, 6)),
+        (elements.Rectangle(0, 0, 5, 5), elements.Rectangle(6, 6, 8, 8)),
+    ],
+)
+def test_separate(rect1, rect2):
+    separate(rect1, rect2)
+
+    assert not rect1.intersects(rect2)
```

Comment:
let's actually also test the new rectangle's coords are what we expected them to be (and modify this test to test all paths in the separate function)